src/frontend/app/features/applications/routes/add-routes/add-routes.component.ts

```diff
@@ -33,7 +33,8 @@ import { Domain } from '../../../../store/types/domain.types';
 import { Route, RouteMode } from '../../../../store/types/route.types';
 import { ApplicationService } from '../../application.service';
 
-
+const hostPattern = '(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])';
```

Comment:
This is not correct - it allows lots of invalid characters - e.g. /, % etc - see: https://en.wikipedia.org/wiki/Hostname for a simple explanation of what is allowed.